src/requests.rs

```diff
@@ -98,208 +104,224 @@ pub async fn get_replays<A, B, C, D, E>(
-                },
-                None => {
-                    return Err(Error::UnexpectedResponse(
-                        "could not split bytes to parse winner",
-                    ))
                 }
-            };
-            //(id_from_bytes(&b[0..18])?, name, winner, time)
-            (id_from_bytes(&b[0..18])?, name, winner, time)
+            }
         }
-        None => {
-            return Err(Error::UnexpectedResponse(
-                "could not find third data part of match",
-            ))
+        Err(e) => {
+            errors.push(ParseError::new(show_buf(bytes), e.into()));
         }
-    };
+    }
 
-    // Construct the match
-    let m = Match {
-        floor: Floor::from_u8(floor)?,
-        timestamp: DateTime::<Utc>::from_utc(
-            NaiveDateTime::parse_from_str(&time, "%Y-%m-%d %H:%M:%S")?,
-            Utc,
-        ),
+    true
+}
+
+fn match_from_replay(replay: messagepack::Replay) -> Result<Match> {
+    Ok(Match {
+        floor: Floor::from_u8(replay.floor)?,
+        timestamp: replay.date,
         players: (
-            Player {
-                id: p1_id,
-                name: p1_name.to_string(),
-                character: Character::from_u8(p1_char)?,
-            },
-            Player {
-                id: p2_id,
-                name: p2_name.to_string(),
-                character: Character::from_u8(p2_char)?,
-            },
+            Player::try_from((replay.player1_character, replay.player1))?,
+            Player::try_from((replay.player2_character, replay.player2))?,
         ),
-        winner: match winner {
+        winner: match replay.winner {
             1 => Winner::Player1,
             2 => Winner::Player2,
             _ => return Err(Error::ParsingBytesError("Could not parse winner")),
         },
+    })
+}
+
+impl TryFrom<(Character, messagepack::Player)> for Player {
+    type Error = Error;
+    fn try_from((character, player): (Character, messagepack::Player)) -> Result<Self> {
+        Ok(Player {
+            id: id_from_bytes(player.id.as_bytes())?,
+            name: player.name,
+            character,
+        })
+    }
+}
+mod messagepack {
+    use super::*;
+
+    use serde_crate::{
+        de::{Deserializer, Error as _},
+        Deserialize,
     };
 
-    Ok(m)
+    use crate::Character;
+
+    impl ReplayRequest {
+        pub fn to_hex(&self) -> String {
+            use std::fmt::Write;
+
+            let mut buf = String::new();
+            for b in rmp_serde::encode::to_vec(self).unwrap() {
+                write!(buf, "{:02X}", b).unwrap();
+            }
+            buf
+        }
+    }
+
+    #[derive(Debug, Clone, Serialize)]
+    #[serde(crate = "serde_crate")]
+    pub struct ReplayRequest {
+        pub header: RequestHeader,
+        pub body: RequestBody,
+    }
+
+    #[derive(Debug, Clone, Serialize)]
+    #[serde(crate = "serde_crate")]
+    pub struct RequestHeader {
+        pub string1: String,
```

Comment:
Any fields that I only know which type they encode, but not what they actually contain just get these generic names. (We have to have something for every field since the data is encoded as nested messagepack arrays)